windows.js

```diff
@@ -62,7 +62,8 @@ async function linkMSYS2() {
 }
 
 export function setupPath(msys2, rubyPrefix) {
-  let path = process.env['PATH'].split(';')
+  const originalPath = process.env['PATH'].split(';')
+  let path = originalPath.slice()
 
   // Remove conflicting dev tools from PATH
   path = path.filter(e => !e.match(/\b(Chocolatey|CMake|mingw64|OpenSSL|Strawberry)\b/))
```

Comment:
@MSP-Greg @slonopotamus What do you think is the best way to proceed?
* Remove this line entirely
* Make it an option to clean to path, and choose a default value
* Remove only `Chocolatey` from this line

What are the risks if we keep any of them in PATH?
Ideally I think a `setup-*` action shouldn't remove anything in PATH except a conflicting tool (Ruby, done below).

Reply:
> What are the risks if we keep any of them in PATH?

Confused people.

> Ideally I think a setup-* action shouldn't remove anything in PATH except a conflicting tool.

I think that would be best.  If people want to use MSYS2, they can use a windows specific action or MSYS2 commands.

But as mentioned, I don't know if a windows specific option to clean the path and enable MSYS2 might be helpful.   It's easy, and that's what the majority of Windows Ruby CI (that requires MinGW compile tools) is doing.

Reply:
> But as mentioned, A windows specific option to clean the path and enable MSYS2 might be helpful. It's easy, and that's what the majority of Windows Ruby CI (that requires MinGW compile tools) is doing.

I think most people expect MSYS2 to be available by default.

And I would think there are cases where one wants both MSYS2 and installing packages with `choco`? Or is that always problematic/a mistake?

With that thinking, if we add an option it would just be an option to clean the PATH, and otherwise leave it untouched (except adding/removing Ruby). I'd still add MSYS2 no matter what, as I don't think there is a need for "no MSYS2" (and therefore "cannot compile C extensions").

Reply:
@slonopotamus For context, what do you want to install with Chocolatey?
Is it something that could be installed with MSYS2's pacman instead?

Do you need to be able to install C extensions on Ruby on Windows?

@MSP-Greg Would installing via pacman generally work better than with Chocolatey for use with Ruby?

Reply:
@eregon 

> Would installing via pacman generally work better than with Chocolatey for use with Ruby?

I've helped a lot of extension gems/repos with Windows CI, and I don't recall ever using a 'Chocolatey package' for compiling.  But, some repos may use it for other requirements.